### PR TITLE
[FLINK-3778] [shell] Forward configuration from FlinkILoop to ScalaShellRemoteStreamEnvironment

### PR DESCRIPTION
**Pull request by tillrohrmann**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [X] General
  - The pull request references the related JIRA issue
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message

- [X] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [X] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

With this PR the configuration of the FlinkILoop is properly forwarded to the
ScalaShellRemoteStreamEnvironment.

### CODE REVIEW

File: flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkILoop.scala [hunk 1]
```diff
@@ -72,7 +72,12 @@ class FlinkILoop(
     
     // create our environment that submits against the cluster (local or remote)
     val remoteBenv = new ScalaShellRemoteEnvironment(host, port, this, clientConfig)
-    val remoteSenv = new ScalaShellRemoteStreamEnvironment(host, port, this);
+    val remoteSenv = new ScalaShellRemoteStreamEnvironment(
+      host,
+      port,
+      this,
+      clientConfig,
+      getExternalJars(): _*);
```
Comment (uce) [inline flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkILoop.scala]: Wow, never seen `: _*` before :smile:

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/RemoteStreamEnvironment.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ public RemoteStreamEnvironment(String host, int port, String... jarFiles) {
 	 * @param port
 	 *            The port of the master (JobManager), where the program should
 	 *            be executed.
-	 * @param config
+	 * @param clientConfiguration
```
Comment (uce) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/RemoteStreamEnvironment.java]: I like the renaming, makes it clearer 👍

**Comment by uce** [comment]
Changes look good to me. +1 to merge (I did not try it out though).

**Comment by tillrohrmann** [comment]
Failing test cases are unrelated. Merging.

**Pull request closed without merging**